File: opium-backend/utils/mailgun.ts
import mailgun from 'mailgun-js';
import Config from "./config";

interface EmailOptions {
  to: string;
  subject: string;
  text?: string;
  html?: string;
}

export const sendEmail = async ({ to, subject, text, html }: EmailOptions): Promise<mailgun.messages.SendResponse> => {
  await Config.load('config.toml');
  const mg = mailgun({
    apiKey: Config.get<string>("mailgun", "api_key"),
    domain: Config.get<string>("mailgun", "domain"),
  });

  const data = {
    from: 'opium <auth@opium.bio>',
    to,
    subject,
    text,
    html,
  };

  return mg.messages().send(data);
};

File: opium-backend-v2/utils/mailgun/mailgun.go
package mailgun

import (
	"context"
	"fmt"
	"time"

	"github.com/mailgun/mailgun-go/v5"
	"github.com/opium-bio/backend-v2/utils/config"
)

func SendEmail(subject, recipient, sender, htmlBody string) error {
	cfg, err := config.Load("config.toml")
	if err != nil {
		return fmt.Errorf("failed to load config: %w", err)
	}
	apiKey := cfg.Mailgun.APIKey
	domain := cfg.Mailgun.Domain
	mg := mailgun.NewMailgun(apiKey)
	message := mailgun.NewMessage(domain, sender, subject, "", recipient)
	message.SetHTML(htmlBody)
	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
	defer cancel()
	resp, err := mg.Send(ctx, message)
	if err != nil {
		return fmt.Errorf("failed to send email: %w", err)
	}
	fmt.Printf("Mailgun response: ID=%s, Message=%s\n", resp.ID, resp.Message)
	return nil
}
